src/main/java/cs361/battleships/models/Board.java

```diff
@@ -182,6 +150,53 @@ private void checkSonarShip(Square s, boolean center) {
 		}
 	}
 
+	void moveFleet(char direction){
+		int count = 0;
+        boolean result = true;
+		for(int i = 0; i < ships.size(); i++){
+			count = 0;
+            boolean thisShip = false;
+            for(int j = 0; j < ships.get(i).getOccupiedSquares().size(); j++) {
+                if (overLaps(ships.get(i).ifMoved(direction, j), ships.get(i))) {
+                    count++;
+                }
+            }
+
+			if(!ships.get(i).moveShip(direction)){
+			    result = false;
+			    thisShip = true;
+            }
+			if(count > 1 && !result && !thisShip){//ships.get(i).getOccupiedSquares().size()) {
+				if(direction == 'N'){
+					ships.get(i).moveShip('S');
+				}else if(direction == 'E'){
+					ships.get(i).moveShip('W');
+				}else if(direction == 'S'){
+					ships.get(i).moveShip('N');
+				}else{
+					ships.get(i).moveShip('E');
```

Comment:
Does this code automatically move ships out of the way of other ships' movement?

Reply:
Yeah this makes it do they do not run into each other.